### PR TITLE
CDAP-15034 CDC solution perf tests

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA - https://issues.cask.co/browse/CDAP-15034
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
- implemented performance tests for "SQL Server -> Bigtable" pipeline

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@jayfjin, could you, please, take a look.

File: src/test/java/co/cask/cdc/plugins/common/BigtableOperations.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Connection;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+public class BigtableOperations {
```
Comment (jayfjin) [inline src/test/java/co/cask/cdc/plugins/common/BigtableOperations.java:31]: javadoc

File: src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class CDCPipelinePT extends CDCPluginPerfTestBase {
```
Comment (jayfjin) [inline src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: Does PT stand for perf test? I'd prefer if we use the full name, CDCPipelinePerfTest

File: src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    long elapsedSeconds = (testEnd - testStart) / 1000;
+    long recordsPerSecond = TEST_DATA_INSERTS / elapsedSeconds;
+    LOG.info("Test finished. Transferred '{}' records. Elapsed time is '{} seconds' ({} records/second)",
+             TEST_DATA_INSERTS, elapsedSeconds, recordsPerSecond);
+  }
+
+  private static int getRowCount(Table table) throws IOException {
+    int rowCount = 0;
+    try (ResultScanner scanner = table.getScanner(new Scan())) {
+      for (Result rs = scanner.next(); rs != null; rs = scanner.next()) {
```
Comment (jayfjin) [inline src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: is there a different function besides scan which can be used to get the row count and is more efficient? If not, prefer while(rs.hasNext()) {
  rs.next();
  rowCount++;
}

File: src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java (hunk 2 above)
Comment (mlozbin-cybervisiontech) [reply src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: in this case I will have to also rename all unit tests to UnitTest becase of class name pattern matching for maven task

File: src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java (hunk 3 above)
Comment (mlozbin-cybervisiontech) [reply src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: there is an AggegationClient for HBase but it is not supported for Bigtable. Also there is no hasNext method for ResultScanner.

Comment (jayfjin) [reply src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: sg

File: src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java (hunk 2 above)
Comment (jayfjin) [reply src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: Prefer that if it's not too much work

Comment (mlozbin-cybervisiontech) [reply src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java]: renamed

File: src/test/java/co/cask/cdc/plugins/common/BigtableOperations.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply src/test/java/co/cask/cdc/plugins/common/BigtableOperations.java:31]: done

**Comment by mlozbin-cybervisiontech** [comment]
@jayfjin addressed all comments

**Pull request merged**